modular_skyrat/master_files/code/modules/client/preferences/headshot.dm

```diff
@@ -0,0 +1,38 @@
+/datum/preference/text/headshot
+	category = PREFERENCE_CATEGORY_NON_CONTEXTUAL
+	savefile_identifier = PREFERENCE_CHARACTER
+	savefile_key = "headshot"
+	/// Assoc list of ckeys and their link, used to cut down on chat spam
+	var/list/stored_link = list()
+
+/datum/preference/text/headshot/apply_to_human(mob/living/carbon/human/target, value, datum/preferences/preferences)
+	target.dna.features["headshot"] = value
+
+/datum/preference/text/headshot/is_valid(value)
+	if(!length(value)) //Just to get blank ones out of the way
+		return TRUE
+	if(!is_veteran_player(usr?.client) && !(usr?.ckey in GLOB.donator_list))
+		to_chat(usr, span_warning("You must be a veteran or donator to use this feature!"))
+		return
+	if(!findtext(value, "https://"))
+		to_chat(usr, span_warning("You need \"https://\" in the link!"))
+		return
+	if(!findtext(value, ".png") && !findtext(value, ".jpg"))
```

Comment:
You should probably have it look for this at the end of the string, not just containing it. 
Because this check would have something like `https://fake.link/the.jpg.exe` be valid

Reply:
Pretty sure this is now resolved